### ISSUE TITLE
Human-Friendly Alerts

### ISSUE THREAD

**Issue by wanderingbort** [eosnetworkfoundation/telegram-bot#2]
In `v0.1.0`, CloudWatch alarms print to Telegram as a very large JSON blob with no whitespace. This is almost useless, especially for someone who does not know what fields to look for. This ticket is to transmute the JSON payload into a human-friendly alert format.

The format should be visually distinctive, using color-coded emoji (✅, ❌, ⚠, etc.) to signify current alarm state. The formatting should be simple enough to convey well via SMS, email, or IM. Finally, the focus should be on reducing debugging time as much as possible by being unambiguous and leaving lots of bread crumbs for engineers to follow to a resolution.

## See Also
**engineering [issue 68](https://github.com/eosnetworkfoundation/engineering/issues/68) - EVM Monitoring and Alerting - Phase 1**
1. eos-evm [issue 602](https://github.com/eosnetworkfoundation/eos-evm/issues/602) - Funnel EVM Health Checks into CloudWatch
1. engineering [issue 48](https://github.com/eosnetworkfoundation/engineering/issues/48) - Collaborate with Operations on Unified Dashboarding Solution
1. engineering [issue 49](https://github.com/eosnetworkfoundation/engineering/issues/49) - Create Bot to Alert via IM on Specific Metrics
1. eos-evm [issue 603](https://github.com/eosnetworkfoundation/eos-evm/issues/603) - **SMS** Alerting for EVM Infrastructure Health Checks
1. engineering [issue 65](https://github.com/eosnetworkfoundation/engineering/issues/65) - **Email** Alerting for EVM Infrastructure Health Checks
1. telegram-bot [issue 1](https://github.com/eosnetworkfoundation/telegram-bot/issues/1) - Open-Source This Repo
1. engineering [issue 57](https://github.com/eosnetworkfoundation/engineering/issues/57) - Create Telegram Service Account
1. engineering [issue 58](https://github.com/eosnetworkfoundation/engineering/issues/58) - Create EVM **Testnet** Alert Channel Using Telegram Service Account
1. engineering [issue 64](https://github.com/eosnetworkfoundation/engineering/issues/64) - Create EVM **Mainnet** Alert Channel Using Telegram Service Account
1. engineering [issue 66](https://github.com/eosnetworkfoundation/engineering/issues/66) - Fix EVM CloudWatch Alerts
1. telegram-bot [issue 2](https://github.com/eosnetworkfoundation/telegram-bot/issues/2) - Human-Friendly Alerts
1. engineering [issue 71](https://github.com/eosnetworkfoundation/engineering/issues/71) - EVM Alerts for APAC Infrastructure
1. telegram-bot [issue 3](https://github.com/eosnetworkfoundation/telegram-bot/issues/3) - Alert Bot Maintainer via Telegram on Errors

**Comment by kj4ezj** [comment]
Telegram notifications are human-friendly! Yaaay!

![2023-10-03 02-59-07 - telegram-bot issue 2](https://github.com/eosnetworkfoundation/telegram-bot/assets/34947245/2ec1f61c-16da-4cf0-96cb-57cc0f4e3f52)

Email and SMS alerts still look like a robot throwing up. Fixing those will require an architecture change, and should perhaps be a separate ticket.

**Comment by kj4ezj** [comment]
## Thursday Update
The transform from JSON to human-friendly notifications was broken out into a separate lambda and repo, [aws-cloudwatch-alarm-handler](https://github.com/eosnetworkfoundation/aws-cloudwatch-alarm-handler). The general architecture is:
> CloudWatch => EventBridge => SNS topic 1 => CloudWatch (formatting) lambda => SNS topic 2 =>
Telegram notification lambda, SMS, email

I stood this entire thing up in parallel to the existing infrastructure so as not to interrupt customer alarms, tested it end-to-end, debugged a bit, and proved it works this evening. This took a bit longer than expected, so I will have to cut releases and put this into production for the EVM team tomorrow.

## Examples
Here is what it looks like.

### Email
View from inbox:

![2023-10-06 01-16-25 - telegram-bot issue 2 example - email preview](https://github.com/eosnetworkfoundation/telegram-bot/assets/34947245/af9b2f22-020c-4a60-ac4f-b5627235e747)

An `INSUFFICIENT_DATA` event.

![2023-10-06 01-17-09 - telegram-bot issue 2 example - email INSUFFICIENT_DATA](https://github.com/eosnetworkfoundation/telegram-bot/assets/34947245/cc10210a-3976-4dfa-9104-1a100065af71)

An `ALARM` event. Note the hyperlink to Telegram.

![2023-10-06 01-17-49 - telegram-bot issue 2 example - email ALARM](https://github.com/eosnetworkfoundation/telegram-bot/assets/34947245/467f84e9-b923-4f59-b9b9-ecd827542939)

An `OK` event.

![2023-10-06 01-18-18 - telegram-bot issue 2 example - email OK](https://github.com/eosnetworkfoundation/telegram-bot/assets/34947245/035f83ac-8305-4171-9e22-cab3dde42eec)


### SMS
Yes, all three images are the same size and no, I do not know why GitHub renders them like this.
INSUFFICIENT_DATA | ALARM | OK
--- | --- | ---
![2023-10-06 01-12-10 - telegram-bot issue 2 example - SMS INSUFFICIENT_DATA - 800p](https://github.com/eosnetworkfoundation/telegram-bot/assets/34947245/789ffcff-1064-4a5b-866c-0c421f2111e7) | ![2023-10-06 01-12-49 - telegram-bot issue 2 example - SMS ALARM - 800p](https://github.com/eosnetworkfoundation/telegram-bot/assets/34947245/07054ca9-1e75-41fe-be0c-01e43d145425) | ![2023-10-06 01-14-37 - telegram-bot issue 2 example - SMS OK - 800p](https://github.com/eosnetworkfoundation/telegram-bot/assets/34947245/ea321f73-fb80-4776-9dff-b5876a1e71fc)

### Telegram
Notice the call to action for the `ALARM` state.

![2023-10-06 01-26-10 - telegram-bot issue 2 example - Telegram notifications](https://github.com/eosnetworkfoundation/telegram-bot/assets/34947245/a135dedf-8f77-4213-be31-94f7c9678d46)

**Comment by kj4ezj** [comment]
## Closing Summary
Human-friendly alerts from `evm-mainnet` and `evm-testnet` infrastructure are being delivered via email, SMS, and Telegram. The [aws-cloudwatch-alarm-handler](https://github.com/eosnetworkfoundation/aws-cloudwatch-alarm-handler) GitHub repo mentioned above was locked down and open-sourced. Stable releases of both the `aws-cloudwatch-alarm-handler` and `telegram-bot` lambdas were cut.

### Architecture
The deployment in each environment consists of [aws-cloudwatch-alarm-handler:v0.1.0](https://github.com/eosnetworkfoundation/aws-cloudwatch-alarm-handler/releases/tag/v0.1.0) and [telegram-bot:v0.3.0](https://github.com/eosnetworkfoundation/telegram-bot/releases/tag/v0.3.0) in this configuration:
> CloudWatch => EventBridge => SNS topic 1 => CloudWatch formatting lambda (`aws-cloudwatch-alarm-handler`) => SNS topic 2 =>
Email, SMS, Telegram notification lambda (`telegram-bot`)

In the event of a runtime error, lambdas attempt to route a notification with a useful error message to me as follows:
> CloudWatch formatting lambda (`aws-cloudwatch-alarm-handler`) => SNS topic **_3_** =>
Email, SMS, Telegram notification lambda (`telegram-bot`)
>
> Telegram notification lambda (`telegram-bot`) => Automation Alerts Telegram channel

For more information, see the `README.md` in the root of each repo, and their respective release notes.
